### ISSUE TITLE
[4.0] possible inline option corner case

### ISSUE THREAD

**Issue by alexlamsl** [jakubpawlowicz/clean-css#866]
I was reading `is-allowed-resource.js` and run into this case:

```
> new CleanCSS({ inline: ['remote', 'local'] }).minify('@import url(http://127.0.0.1/remote.css);', console.log);
null { styles: '@import url(http://127.0.0.1/remote.css);',
  stats:
   { efficiency: 0,
     minifiedSize: 41,
     originalSize: 41,
     timeSpent: 4 },
  errors: [],
  inlinedStylesheets: [],
  warnings: [ 'Skipping remote @import of "http://127.0.0.1/remote.css" as resource is not allowed.' ] }
```

Admittedly `['remote', 'local']` should probably just be `['all']`, so I'm not sure if this is a bug or not.

**Comment by alexlamsl** [comment]
Unrelated: when I run `npm test` locally (on Windows) I don't seem to be exercising `test/protocol-imports-test.js`

**Comment by jakubpawlowicz** [comment]
This looks like a bug to me!

Regarding `protocol-imports-test.js`, could you comment out the following lines and let me know if tests pass OK? https://github.com/jakubpawlowicz/clean-css/blob/master/test/protocol-imports-test.js#L14-L15

I remember having issues with those tests as they use `nock` but also spin off servers to test remote imports.

**Comment by alexlamsl** [comment]
Ah, that explains it - will test and report back when I get back to my desk in an hour or so.

(Note to self: read from top of file next time before typing...)

BTW, the reason I was reading those source files is wondering whether `{"local": true, "remote": function(uri, callback) {/*do your request magic*/}}` would solve issues like #755

**Comment by jakubpawlowicz** [comment]
Thanks!

I haven't looked into #755 yet but it looks like sth we could handle internally.

**Comment by alexlamsl** [comment]
Tested on Node.js 7.4.0 with those lines removed, and `protocol-imports-test.js` passed successfully :wink: 

(It failed with `node-chakracore` and I've filed https://github.com/nodejs/node-chakracore/issues/170 for them to fix that up.)

**Comment by jakubpawlowicz** [comment]
Seems to work fine indeed, pushed to master in 05e792deb93f5236916113df1d29655776c37b1c

**Comment by jakubpawlowicz** [comment]
Fixed in 4.0.1.
